runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -115,14 +115,14 @@ function LanguageTree:parse()
       parser:set_included_ranges(ranges)
 
       local tree, tree_changes = parser:parse(old_tree, self._source)
-      self:_do_callback('changedtree', tree_changes, tree)
+      self:_do_callback('changedtree', tree_changes, tree, self._lang)
 
       table.insert(self._trees, tree)
       vim.list_extend(changes, tree_changes)
     end
   else
     local tree, tree_changes = parser:parse(old_trees[1], self._source)
-    self:_do_callback('changedtree', tree_changes, tree)
+    self:_do_callback('changedtree', tree_changes, tree, self._lang)
```

Comment:
Probably some of the other callbacks could also profit from having the language. So it could be predicatable that wen you get the parsetree you would also get for which language the parseresult is because in most situations you can't do that much with a parse result when you don't know the language. But you can ask about opinions first.